data_diff/__init__.py

```diff
@@ -1,9 +1,86 @@
-from typing import Tuple
+from typing import Tuple, Iterator, Optional
 
 from .database import connect_to_uri
-from .diff_tables import TableSegment, TableDiffer
+from .diff_tables import (
+    TableSegment,
+    TableDiffer,
+    DEFAULT_BISECTION_THRESHOLD,
+    DEFAULT_BISECTION_FACTOR,
+    DbKey,
+    DbTime,
+    parse_table_name,
+)
 
 
-def create_source(db_uri: str, table_name: str, key_column: str, extra_columns: Tuple[str, ...] = ()):
-    db = connect_to_uri(db_uri)
-    return TableSegment(db, (table_name,), key_column, tuple(extra_columns))
+class TableRef:
+    """Connects to a URI and creates a table ref
+
+    This is an auxiliary class for diff_tables()
+    """
+
+    def __init__(self, db_uri: str, table_name: str):
+        self.db = connect_to_uri(db_uri)
+        self.table_name = table_name
+
+    def create_table_segment(self, **kwargs):
+        return TableSegment(self.db, parse_table_name(self.table_name), **kwargs)
+
+
+def diff_tables(
+    table1: TableRef,
+    table2: TableRef,
+    *,
+    # Name of the key column, which uniquely identifies each row (usually id)
+    key_column: str = "id",
+    # Name of updated column, which signals that rows changed (usually updated_at or last_update)
+    update_column: str = None,
+    # Extra columns to compare
+    extra_columns: Tuple[str, ...] = (),
```

Comment:
I kind of think this should just be "columns" in the user-facing API, but I don't feel too strongly

Reply:
I think `columns` might imply that it overrides `key_column` or `update_column`, or that you need to specfy them twice.